src/core/server/rendering/bootstrap/bootstrap_renderer.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createHash } from 'crypto';
+import * as UiSharedDeps from '@kbn/ui-shared-deps';
+import { PackageInfo } from '@kbn/config';
+import { UiPlugins } from '../../plugins';
+import { IUiSettingsClient } from '../../ui_settings';
+import { HttpAuth, KibanaRequest } from '../../http';
+import { getPluginsBundlePaths } from './get_plugin_bundle_paths';
+import { BootstrapTemplateInterpolator } from './template_interpolator';
+
+export type BootstrapRendererFactory = (factoryOptions: FactoryOptions) => BootstrapRenderer;
+export type BootstrapRenderer = (options: RenderedOptions) => Promise<RendererResult>;
+
+interface FactoryOptions {
+  serverBasePath: string;
+  packageInfo: PackageInfo;
+  uiPlugins: UiPlugins;
+  auth: HttpAuth;
+}
+
+interface RenderedOptions {
+  request: KibanaRequest;
+  uiSettingsClient: IUiSettingsClient;
+}
+
+interface RendererResult {
+  body: string;
+  etag: string;
+}
+
+export const bootstrapRendererFactory: BootstrapRendererFactory = ({
+  packageInfo,
+  serverBasePath,
+  uiPlugins,
+  auth,
+}) => {
+  const templateInterpolator = new BootstrapTemplateInterpolator();
+
+  const isAuthenticated = (request: KibanaRequest) => {
+    if (!auth.isEnabled()) {
+      return true;
+    }
+    const { status: authStatus } = auth.get(request);
+    // status is 'unknown' when auth is disabled. we just need to not be `unauthenticated` here.
+    return authStatus !== 'unauthenticated';
+  };
```

Comment:
This `isAuthenticated` function could be renamed `amIAuthorizedToQueryElasticsearch`, as it returns true either when `auth` is disabled (no security), or if `auth` is enabled AND the user is authenticated.

(Long) note: now that we moved the stylesheet loading to `RenderingServiceSetup.render`, the only remaining reason to load uiSettings stored in ES when rendering `bootstrap.js` is to compute the  `__kbnThemeTag__` (`themeTag` below) variable (that is used as a prefix by our client-side loader for stylesheets). I tried to move that variable initialization to the rendered document instead, however, because of our CSP policy, we can't use inline scripts (and injecting the variable from the client-side `core` code is too late)

An improvement would be to (re-)add a `nonce-XXX` option to our CSP, to let the rendering service have inlined scripts, but doing that properly (aka not as it was done in legacy before we removed it) would require to generate the `nonce` value per-request and to have the rendering service access this value when rendering the document. If technically doable, it is not trivial and require some thinking, so imho (given that we really want that) it should be done as a follow-up (we would even be able to inline the whole `bootstrap.js` script in the rendered document in that case). 

cc @legrego @joshdover this is the reason why I was speaking about reintroducing `nonce-`

Reply:
Created https://github.com/elastic/kibana/issues/93785